src/platform/plugins/shared/controls/public/controls/data_controls/options_list_control/components/options_list_popover_suggestions.test.tsx

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import React from 'react';
+
+import { act, fireEvent, render } from '@testing-library/react';
+
+import { take } from 'lodash';
+import { getOptionsListMocks } from '../../mocks/api_mocks';
+import { MAX_OPTIONS_LIST_REQUEST_SIZE, MIN_OPTIONS_LIST_REQUEST_SIZE } from '../constants';
+import { ContextStateManager, OptionsListControlContext } from '../options_list_context_provider';
+import type { OptionsListComponentApi } from '../types';
+import { OptionsListPopoverSuggestions } from './options_list_popover_suggestions';
+
+describe('Options list popover', () => {
+  const waitOneTick = () => act(() => new Promise((resolve) => setTimeout(resolve, 0)));
+
+  const allOptions = [
+    { value: 'moo', docCount: 1 },
+    { value: 'tweet', docCount: 2 },
+    { value: 'oink', docCount: 3 },
+    { value: 'bark', docCount: 4 },
+    { value: 'meow', docCount: 5 },
+    { value: 'woof', docCount: 6 },
+    { value: 'roar', docCount: 7 },
+    { value: 'honk', docCount: 8 },
+  ];
+
+  const mountComponent = ({
+    api,
+    displaySettings,
+    stateManager,
+    showOnlySelected,
+  }: {
+    api: any;
+    displaySettings: any;
+    stateManager: any;
+    showOnlySelected?: boolean;
+  }) => {
+    return render(
+      <OptionsListControlContext.Provider
+        value={{
+          api: api as unknown as OptionsListComponentApi,
+          displaySettings,
+          stateManager: stateManager as unknown as ContextStateManager,
+        }}
+      >
+        <OptionsListPopoverSuggestions showOnlySelected={showOnlySelected ?? false} />
+      </OptionsListControlContext.Provider>
+    );
+  };
+
+  test('displays "load more" text when possible', async () => {
+    const mocks = getOptionsListMocks();
+    mocks.api.totalCardinality$.next(allOptions.length);
+    mocks.api.availableOptions$.next(take(allOptions, 5));
+    const suggestionsComponent = mountComponent(mocks);
+
+    // the cardinality is larger than the available options, so display text
+    let optionComponents = await suggestionsComponent.findAllByRole('option');
+    expect(optionComponents.length).toBe(6);
+    expect(
+      suggestionsComponent.queryByTestId('optionsList-control-selection-honk')
+    ).not.toBeInTheDocument();
+    expect(suggestionsComponent.queryByTestId('optionslist--canLoadMore')).toBeInTheDocument();
+
+    // we are displaying all the available options - so don't display "load more" text
+    mocks.api.availableOptions$.next(allOptions);
+    await waitOneTick();
```

Comment:
Instead of `waitOneTick`, you can just use `waitFor`. This will eliminate any race conditions with timing. For example, if rendering options takes more then one tick, the test will still work since its waiting until there are 9 options instead of waiting a static period of time.

```
await waitFor(() => {
  optionComponents = await 
  suggestionsComponent.findAllByRole('option');
    expect(optionComponents.length).toBe(9);
});
```

Reply:
Good call! Done in 2f853769ada6de3dec591d419aaf5398042232bd :+1: